expression/integration_test.go

```diff
@@ -2469,21 +2469,34 @@ func (s *testIntegrationSuite) TestDateBuiltin(c *C) {
 	r = tk.MustQuery("select date'2017/12-12'")
 	r.Check(testkit.Rows("2017-12-12"))
 
-	r = tk.MustQuery("select date'0000-00-00'")
-	r.Check(testkit.Rows("<nil>"))
+	r = tk.MustQuery("select date '0000-00-00';")
+	r.Check(testkit.Rows("0000-00-00"))
```

Comment:
in mysql, this will cause an error.

Reply:
@XuHuaiyu What's your mysql version? It works on my computer.
```sql
# ./mysql -uroot -pfreeze3131
mysql: [Warning] Using a password on the command line interface can be insecure.
Welcome to the MySQL monitor.  Commands end with ; or \g.
Your MySQL connection id is 35
Server version: 5.7.17-log MySQL Community Server (GPL)

Copyright (c) 2000, 2016, Oracle and/or its affiliates. All rights reserved.

Oracle is a registered trademark of Oracle Corporation and/or its
affiliates. Other names may be trademarks of their respective
owners.

Type 'help;' or '\h' for help. Type '\c' to clear the current input statement.

mysql> select date '0000-00-00';
+-------------------+
| date '0000-00-00' |
+-------------------+
| 0000-00-00        |
+-------------------+
1 row in set (0.00 sec)

mysql> 

```

Reply:
@XuHuaiyu The result seems has something to do with `SQL_MODE`. I'll fix this.